lib/commands/getNewAddress.js

```diff
@@ -0,0 +1,91 @@
+const path = require('path');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+const compose = require('docker-compose');
+const waitForCoreStart = require('../util/waitForCoreStart');
+const waitForCoreSync = require('../util/waitForCoreSync');
+const generateLocalBlocks = require('../util/generateLocalBlocks');
+
+const SATOSHI_MULTIPLIER = 10 ** 8;
+
+/**
+ * Generate new address with requested amount of dash
+ *
+ * @param {string} preset
+ * @param {number} amount
+ * @returns {Promise<void>}
+ */
+async function getNewAddress(preset, amount) {
+  let address;
+  let privateKey;
+
+  const composeOptions = {
+    cwd: path.join(__dirname, '../'),
+    config: 'docker-compose.yml',
+    composeOptions: [
+      '--env-file', `../.env.${preset}`,
+    ],
+  };
+
+  try {
+    // start dash core
+    await compose.upOne('core', composeOptions);
+
+    const dashcoreConfig = {
+      protocol: 'http',
+      user: 'dashrpc',
+      pass: 'password',
+      host: '127.0.0.1',
+      port: 20002,
+    };
+
+    const coreClient = new RpcClient(dashcoreConfig);
+
+    // wait dash core to start
+    await waitForCoreStart(coreClient);
+
+    // wait dash core to be synced
+    if (preset !== 'local') {
+      await waitForCoreSync(coreClient);
+    }
+
+    ({ result: address } = await coreClient.getNewAddress());
```

Comment:
Let's use dashcore-lib for that.

Reply:
If we want to use this address to register masternode later, we'll need to import this private key into the core wallet. And it takes some time because the wallet will reindex all blockchain. So in this case, such a way of generation would be faster.